### ISSUE TITLE
No namespaces/files shows up in generated HTML

### ISSUE THREAD

**Issue by danr1979** [phpDocumentor/phpDocumentor2#1523]
Hi.

I am new to phpdocumenter and despite reading the documentation and searching the Web for help I cannot get any content to appear in the HTML generated by phpdocumenter.

When I run the phpdoc command using:

```
./phpdoc -d mysite -t /phpdoc-site
```

all of my files seem to be parsed and generation succeeds.  However, on visiting the HTML generated I just see the default template with a \ in the left-hand bar and a \ under the "namespace hierarchy".

Many of my source files are namespaced but not all of them. An example of how my source files are written is:

```
<?php namespace My\Files;

use Paya\Libraries\Signup;

/** 
 * My summary
 * 
 * My description
 *
 * @author  Me <me@test.com>
 * 
 */
class MyClass
{
```

I don't currently have a file doc block but from what I have read this is fine.  Currently only one of my files has function doc blocks but I would have thought I would at least see the class documented in the generated HTML.

I have tried using the -force switch on the command line when generating the documentation but this hasn't helped.

Any help or suggestions would be appreciated.

Thank you.

**Comment by danr1979** [comment]
Hm, my problem could be this:

https://github.com/phpDocumentor/phpDocumentor2/issues/1427

so I'll try to install the phar or pear version to see if that fixes it.

**Comment by danr1979** [comment]
Yep that fixed it.  There appears to be a bug in the composer install.  I was using phpDocumenter 2.8.2 so the issue still persists in that version.

**Comment by nicjansma** [comment]
I hit this issue today as well and spent a couple hours trying to figure out why my phpdoc output was empty.  The PHAR version works.

It would be nice if you spun a 2.8.3 version to alleviate this for composer users.

**Comment by beporter** [comment]
[As mentioned](https://github.com/phpDocumentor/phpDocumentor2/issues/1427#issuecomment-123042748) in #1427, even the 2.8.4 composer package release still seems to exhibit this issue. The phar (downloaded today, reporting `v2.8.2` does seem to work properly though.)

**Comment by beporter** [comment]
This issue continues to persist in the `2.8.5` composer release for me.

**Comment by KyraD** [comment]
I'm also having this issue, with **both** composer (2.8.5) and phar (2.8.5).

@beporter Did you try with latest phar? Also, do you know where I can download the 2.8.2 version you tested with that works? I could only find the latest available for download.

**Comment by beporter** [comment]
@KyraD No, I have not tested with the latest phar. You can probably try the [Releases](https://github.com/phpDocumentor/phpDocumentor2/releases) page for older versions of the file, although I admit I haven't tried to actually download and test any of those.

(Also remember that the version numbers reported by the phar's themselves via `phpdocumentor.phar --version` doesn't necessarily always correctly match the released version number. In other words, if you download the 2.8.3 phar, it may report itself as being 2.8.2 still.)

**Comment by quantumpacket** [comment]
@KyraD I just tested using phar version being reported as 2.8.2 and it does not work for me either. I'm testing under PHP7. Hopefully this gets fixed soon.

**Comment by mbed67** [comment]
@KyraD may I ask what version of nikic/php-parser you have? I just cloned a fresh copy of phpDocumentor2, ran composer install and I got nikic/php-parser v.0.9.5
With this version all works well. But when I force my dependency of phpDocumentor/reflection to an older version (1.0.3) then I get nikic/php-parser v1.4.0. Then it doesn't work. The version of nikic/php-parser has to be v.0.9.5.

**Comment by KyraD** [comment]
Deleted `vendor` directory and `composer.lock` then did fresh install and tried, did not work. When I navigate to `./vendor/nikic/php-parser/composer.json` I get the following:

```json
{
    "name": "nikic/php-parser",
    "description": "A PHP parser written in PHP",
    "keywords": ["php", "parser"],
    "type": "library",
    "license": "BSD-3-Clause",
    "authors": [
        {
            "name": "Nikita Popov"
        }
    ],
    "require": {
        "php": ">=5.2",
        "ext-tokenizer": "*"
    },
    "autoload": {
        "psr-0": { "PHPParser": "lib/" }
    },
    "extra": {
        "branch-alias": {
            "dev-master": "0.9-dev"
        }
    }
}
```
My composer file has the following settings if it helps:

```json
    "require": {
        "php": "~5.5|~7.0"
    },
    "require-dev": {
        "phpdocumentor/phpdocumentor": "~2.8",
        "phpmd/phpmd": "~2.2",
        "phpunit/phpunit": "~4.8",
        "squizlabs/php_codesniffer": "~2.3"
    },
    "minimum-stability": "RC",
    "prefer-stable": true,
    "config": {
        "optimize-autoloader": true,
        "preferred-install": "dist"
    }
```

**Comment by mbed67** [comment]
@KyraD Could you tell me what the version number is of nikic/php-parser in the composer.lock?

**Comment by KyraD** [comment]
```json
{
    "name": "nikic/php-parser",
    "version": "v0.9.5",
    "source": {
        "type": "git",
        "url": "https://github.com/nikic/PHP-Parser.git",
        "reference": "ef70767475434bdb3615b43c327e2cae17ef12eb"
    },
    "dist": {
        "type": "zip",
        "url": "https://api.github.com/repos/nikic/PHP-Parser/zipball/ef70767475434bdb3615b43c327e2cae17ef12eb",
        "reference": "ef70767475434bdb3615b43c327e2cae17ef12eb",
        "shasum": ""
    },
    "require": {
        "ext-tokenizer": "*",
        "php": ">=5.2"
    },
    "type": "library",
    "extra": {
        "branch-alias": {
            "dev-master": "0.9-dev"
        }
    },
    "autoload": {
        "psr-0": {
            "PHPParser": "lib/"
        }
    },
    "notification-url": "https://packagist.org/downloads/",
    "license": [
        "BSD-3-Clause"
    ],
    "authors": [
        {
            "name": "Nikita Popov"
        }
    ],
    "description": "A PHP parser written in PHP",
    "keywords": [
        "parser",
        "php"
    ],
    "time": "2014-07-23 18:24:17"
}
```

**Comment by mbed67** [comment]
@KyraD Okay, thank you. I'll have to look further then, but I won't be able to do that until tomorrow when I have my day off from work. I'll see if I can reproduce it tomorrow.

**Comment by mbed67** [comment]
Hi @KyraD, unfortunately I'm not able to reproduce the issue. This is what I have tried:
Using php 5.5.12 I created an empty directory with only your composer.json. Then I ran composer.install. So now I should have the same configuration as you have.
From that directory I ran:
.\vendor\bin\phpdoc -d .\vendor\phpdocumentor\phpdocumentor\src\phpDocumentor\Descriptor -t .\docs\clean (I'm on windows right now, so the slashes are backwards, however my try from last wednesday was on ubuntu and that worked all right too)
As you can see, I'm testing on files of phpDocumentor itself.
This went well. 
I also tried it with php 7.0.0 RC2 and this also went well. All links are clickable and I can reach the file info.
![phpdoc](https://cloud.githubusercontent.com/assets/7139949/9681612/a54740ee-52ff-11e5-9dbc-102b4a9a4215.JPG)

**Comment by mbed67** [comment]
@KyraD what happens if you use another template, for instance add --template="responsive-twig" to the command?

**Comment by KyraD** [comment]
@mbed67 I upgraded to 7.0.0RC2 and now vendor packages get documented. They were not before. Dunno if it was an issue with PHP or something else. However, my own package still does not document. `./src/CompanyName` is not showing up, but  `./vendor` and `./tests/unit/CompanyNameTest` are being documented. I'm running phpdoc on the parent directory. I'm going to dig a little more and see what's up, I'm sure I'm doing something really silly =oP Unfortunately I can't post the files since they are internal files for server administration. Thanks for the help so far! =o)

**Comment by beporter** [comment]
Just wanted to chime in: For me the issue is occurring on PHP 5.4.x through 5.6.x.

**Comment by mbed67** [comment]
@KyraD I'm glad you are getting a step further now. And I hope you'll soon find out why your own package doesn't document. If you do, it would be a great help for others if you could share your solution in this discussion. Have a good day :-)

**Comment by Kryten0807** [comment]
For what it's worth, I'm experiencing this with phpdoc version 2.8.5 on php 5.5.9-1ubuntu4.14.

the `.phar` version works (also version 2.8.5)

**Comment by jaapio** [comment]
Closing as this is reported on phpdoc v2 which is unmaintained
